[FIX] website_sale: prevent issue with test shop_list_view_b2c

seb-odoo opened the pull request:
It very rarely happens that the test is failing due to the following error:

`Tour shop_list_view_b2c failed at step
.oe_product_cart .oe_currency_value:contains("825.00")`

The value found in that case is 750 instead of 825, which indicates that the b2c
setting was not correctly activated.

To exclude the RPC themselves from the equation, it is decided here to move the
setting change directly in the Python.

seb-odoo commented:
@rdeodoo could you double check this please

@robodoo delegate=rdeodoo

cc @d-fence

rdeodoo commented:
It was done that way so the tour can be done from console without going through required python code, right?
I guess we can move it to python, as some tours have already been changed that way anyway..
But the runbot is red, the change is probably not enough.

seb-odoo commented:
This particular tour already depends on data created in `setUp` so I don't think it is making it worse to remove the RPC in regard to starting it from the browser.

But I see that runbot is red indeed, it doesn't like my extra assertion (that is working fine locally). I'll investigate that, already made a rebuild to see if the issue is consistent.

seb-odoo commented:
I removed the extra assertion I added from the python.

Its goal was that if the non deterministic issue happens again it would fail as early as possible.

But to make that extra assertion reliable it would take more code (especially correct pricelist selection).
But it is unnecessary to add that code since it basically asserts the same thing as the first step of the JS.

So let's just see in the long run if the non deterministic issue happens again after this change.

rdeodoo commented:
is this a real race condition btw? I don't see it listed in http://runbot.odoo.com/web#action=573&model=runbot.build.error&view_type=list&cids=&menu_id=405

rdeodoo commented:
Otherwise it LGTM, waiting d-fence insight before r+?

seb-odoo commented:
He pinged me for it, the "Build errors" list cannot be filled with errors from v12.0 because the format is not the same, but you can find it inside "Logs/Error Logs" (5 occurrences) searching the error message.

rdeodoo commented:
@robodoo r+ then :crossed_fingers:
